pkg/kubelet/cri/remote/remote_runtime.go

```diff
@@ -624,20 +624,21 @@ func (r *remoteRuntimeService) containerStatusV1(ctx context.Context, containerI
 }
 
 // UpdateContainerResources updates a containers resource config
-func (r *remoteRuntimeService) UpdateContainerResources(containerID string, resources *runtimeapi.LinuxContainerResources) (err error) {
+func (r *remoteRuntimeService) UpdateContainerResources(containerID string, resources *runtimeapi.ContainerResources) (err error) {
 	klog.V(10).InfoS("[RemoteRuntimeService] UpdateContainerResources", "containerID", containerID, "timeout", r.timeout)
 	ctx, cancel := getContextWithTimeout(r.timeout)
 	defer cancel()
 
 	if r.useV1API() {
 		_, err = r.runtimeClient.UpdateContainerResources(ctx, &runtimeapi.UpdateContainerResourcesRequest{
 			ContainerId: containerID,
-			Linux:       resources,
+			Linux:       resources.GetLinux(),
+			Windows:     resources.GetWindows(),
 		})
 	} else {
 		_, err = r.runtimeClientV1alpha2.UpdateContainerResources(ctx, &runtimeapiV1alpha2.UpdateContainerResourcesRequest{
 			ContainerId: containerID,
-			Linux:       v1alpha2LinuxContainerResources(resources),
+			Linux:       v1alpha2LinuxContainerResources(resources.GetLinux()),
```

Comment:
We updated v1alpha2 with the new ContainerResources structure, so I think we should do the same we do above. If we don't update v1alpha2, then we can leave this as it is now IMO

Reply:
yes, adding..

Reply:
There is no conversion function for v1alpha2LinuxContainerResources.
Is it wise to just add an unsafe pointer conversion func? I don't have a way to test/verify.
I'm not very sure about this below. Does it look right to you?
```diff
diff --git a/pkg/kubelet/cri/remote/conversion.go b/pkg/kubelet/cri/remote/conversion.go
index 35539397755..120b718cbf5 100644
--- a/pkg/kubelet/cri/remote/conversion.go
+++ b/pkg/kubelet/cri/remote/conversion.go
@@ -113,6 +113,10 @@ func v1alpha2LinuxContainerResources(from *runtimeapi.LinuxContainerResources) *
        return (*v1alpha2.LinuxContainerResources)(unsafe.Pointer(from))
 }
 
+func v1alpha2WindowsContainerResources(from *runtimeapi.WindowsContainerResources) *v1alpha2.WindowsContainerResources {
+       return (*v1alpha2.WindowsContainerResources)(unsafe.Pointer(from))
+}
+
 func v1alpha2ExecRequest(from *runtimeapi.ExecRequest) *v1alpha2.ExecRequest {
        // If this function changes, also adapt the corresponding Exec code in
        // pkg/kubelet/cri/remote/remote_runtime.go
diff --git a/pkg/kubelet/cri/remote/remote_runtime.go b/pkg/kubelet/cri/remote/remote_runtime.go
index a09879b346a..976f845e011 100644
--- a/pkg/kubelet/cri/remote/remote_runtime.go
+++ b/pkg/kubelet/cri/remote/remote_runtime.go
@@ -639,6 +639,7 @@ func (r *remoteRuntimeService) UpdateContainerResources(containerID string, reso
                _, err = r.runtimeClientV1alpha2.UpdateContainerResources(ctx, &runtimeapiV1alpha2.UpdateContainerResourcesRequest{
                        ContainerId: containerID,
                        Linux:       v1alpha2LinuxContainerResources(resources.GetLinux()),
+                       Windows:     v1alpha2WindowsContainerResources(resources.GetWindows()),
                })
        }
        if err != nil {
```

Reply:
Perhaps @marosset can comment if this is needed.

Reply:
yeah that looks correct. this could also take the approach of other recent CRI changes and just drop v1alpha2 support

Reply:
or rather, not support Windows in v1alpha2, and not add the additional ContainerResources. Thinking about it more (because we already use UpdatecontainerResources for the cpumanager) we may just want to add support in v1alpha2 to make sure we don't regress

Reply:
SG. Adding with rebase fix.

Reply:
fixed

Reply:
It looks like containerd has had support updating Windows resources in `UpdateContainerResources` since the 1.6 release (which I'm pretty sure still used `v1alpha2`).
Because of this I think we should include the Windows changes in `v1alpha2` and `v1`.
x-ref https://github.com/containerd/containerd/pull/5778

Reply:
Actually the changes I just linked use the `v1` API.
I don't think we need to support `v1alpha2` but since the changes are already in might as well leave them?

Reply:
We can remove it with the main feature changes in 1.26. I've added a [TODO tracking item](https://github.com/kubernetes/kubernetes/issues/109547) for this.